### DISCUSSION TITLE
如何讓你code充滿文學氣息

### DISCUSSION THREAD

**Discussion by takoooo1201** [1101-GS4719-C/Python-Community-GS4719-C#1933] [General]
# 如何讓你code充滿文學氣息
常寫程式的你一定有這個困擾，覺得程式碼索然無味，少了一點藝術的美，這篇文章會教如何讓你的程式碼充滿文學氣息，整個直接文青起來。

1. write your header file
Write a header file to hide your macros, and use #define directives to define it. you can write anything to replace your original code, e.g. "#define 白日依山盡 int main(){"
p.s. Macros in programming are shortcuts for writing repetitive code or defining constants. They are defined using the #define directive and are processed by the preprocessor before actual compilation.

2. write your main C source file
includes the xxx.h header file in your main source file. This allows you to use the macros defined in xxx.h within your main source file.

3. compile your c code
Using Ubuntu as an example.
- gcc ./fileName.c 
//compile fileName.c using the GNU Compiler Collection (GCC) and generates an executable file, typically named fileName.out by default.
- ./fileName.out
//run the executable file

4. final
c code中明明是一首詩但卻可執行並印出hello world
![圖片](https://github.com/NCU-GS4719-Python/Python-Community/assets/160357236/f919e1e0-ddfe-48ab-b1c8-f1e6805ac4da)

5. Reflection:
我覺得以一個常在寫code的人來說將程式码注入文學氣息的想法很有趣。我常常覺得程式設計有時會缺乏其他形式寫作中所具有的藝術表達。 這篇文章提供了一個創意的方法來解決這個問題。
我認為使用頭檔來封裝巨集的建議是一個聰明的辦法，可以隱藏程式碼的複雜性。 透過在巨集中使用詩意的表達方式，例如 "#define 白日依山盡 int main(){"，程式碼變得更具想像力和感染力。 這種方法不僅增加了藝術美感，也使程式碼更加引人入勝和令人難忘。
整體來說，我認為將程式碼注入文學氣息的概念是一種令人耳目一新、創新的方法，可以讓程式設計更具表現力和樂趣。 它鼓勵程式設計師創造性地思考，並探索透過程式碼表達自己的新方式。 我期待在自己的專案中嘗試這種方法，並探索它提升程式碼美學吸引力的潛力。

**Comment by lyykey** [comment]
筆者對於文學氣息有一定的追求:)，對我而言，我反而會存放一些 language 語法，在有些考試的時候，作為輔助工具哈哈，不過也是感謝筆者分享這個資訊。

**Comment by benbebobo149** [comment]
我蠻推薦你使用文言文程式語言，除了增加你的文言文的撰寫之外，更可以讓中文系的女生對你刮目相看。換一個語言，也等於換了一個思考方式，或許透過這個程式語言，可以激盪出新的創意。我覺得這種嘗試挑戰自己的精神很值得鼓勵，而且可能會讓你在程式設計的路上更上一層樓。
以下是其repo的連結。
https://github.com/wenyan-lang/wenyan

**Comment by yubo0oo** [comment]
我覺得，將程式碼注入文學氣息的想法真的是太有趣了！對於常寫程式的我來說，這樣的方法不僅增加了程式碼的藝術美感，也讓程式設計變得更加生動有趣。程式設計有時會缺乏其他形式寫作中的藝術表達，而這篇文章提供了一個創意的方法來解決這個問題。

使用頭檔來封裝巨集的建議真的很聰明，這樣可以隱藏程式碼的複雜性。透過在巨集中使用詩意的表達方式，例如 "#define 白日依山盡 int main(){"，程式碼變得更具想像力和感染力。這種方法不僅增加了藝術美感，也使程式碼更加引人入勝和令人難忘。

整體來說，我覺得將程式碼注入文學氣息的概念是一種令人耳目一新、創新的方法，讓程式設計更具表現力和樂趣。這不僅鼓勵程式設計師創造性地思考，還能探索透過程式碼表達自己的新方式。我非常期待在自己的專案中嘗試這種方法，並探索它提升程式碼美學吸引力的潛力。

不知道大家對這個想法有什麼看法呢？也歡迎分享你們的經驗和感受！

**Comment by poyen0806** [comment]
我很想推薦你嘗試一下文言文程式語言，這是一種將程式設計與古典文學結合的有趣方式。不僅充滿文學氣息，還能讓你在寫程式時彷彿置身於古代書院中，體驗一種別具一格的編碼感受！

最妙的是，想抄你的 code 的人，還得先看得懂文言文，直接多了一層防禦機制，別有一番趣味。對，像我就看不懂這東西，我國文沒那麼好。

除此之外，學習文言文程式語言也是一種挑戰自己邏輯思維與文學造詣的絕佳機會，讓你的程式作品更富有文化底蘊。對於喜歡文學與程式設計的人來說，這絕對是值得嘗試的特別體驗！

這裡是它的 GitHub repo，讓你一窺它的魅力：
https://github.com/wenyan-lang/wenyan
